cmd/kubelet/app/server.go

```diff
@@ -423,9 +424,13 @@ func Run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 	return nil
 }
 
+// checkPermissions confirms that, on linux, the UID of the kubelet process is 0.  For windows, it will return an error automatically.  For other systems, the behaviour is unspecified.
 func checkPermissions() error {
+	if runtime.GOOS == "windows" {
```

Comment:
How about doing like
```
if uid := os.Getuid(); uid != 0 {
    if runtime.GOOS == "windows" {
        return fmt.Errorf("failed to check UID permissions because of Windows")
    }
    return fmt.Errorf("kubelet needs to run as uid `0`. It is being run as %d", uid)
```
for understanding easily the function returns -1 on Windows.

Reply:
this check is only meaningfull on linux.  because there is no uid on windows... does that make sense?

Reply:
actually i think nvm i ll implement your idea , its not a big deal either way :)

Reply:
fixed